lib/assets/javascripts/views/devise/registrations/edit.js

```diff
@@ -17,4 +18,30 @@ $(() => {
     $('#org-controls .combobox-clear-button').hide();
     $('#other_org_toggle a').hide();
   }
+
+  // If the user changes their email address display the password
+  // confirmation modal
+  $('#personal_details_registration_form #user_email').blur((e) => {
+    if (isObject($('#original_email')) && isObject($(e.currentTarget))) {
+      const original = $('#original_email').val();
+      const email = $(e.currentTarget).val();
+
+      if (isString(original) && isString(email)) {
+        if (original.toLowerCase() !== email.toLowerCase()) {
+          $('#password-confirmation').modal('show');
+        }
+      }
+    }
+  });
+
```

Comment:
what about pressing Escape or clicking outside the region enabled for the popup? You will need the body of this click handler function too...